File: CWM.Services/Models/Part.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CWM.Database.Models
{
    [Table("Parts")]
    public class Part
    {
        [Key]
        public int Id { get; set; }

        public string SerialNumber { get; set; } = string.Empty;
        public string Manufacturer { get; set; } = string.Empty;
        public string PartName { get; set; } = string.Empty;
        public byte[]? Image { get; set; }
        public decimal Price { get; set; }
        public string Description { get; set; } = string.Empty;

        public virtual ICollection<PartWorkOrder> PartWorkOrder { get; set; } = new List<PartWorkOrder>();
    }
}

File: CWM.UI/cwm_desktop_mobile/lib/models/part.dart
import 'package:json_annotation/json_annotation.dart';

part 'part.g.dart';

@JsonSerializable()
class Part {
  int id;
  String serialNumber;
  String manufacturer;
  String partName;
  String? image;
  double price;
  String description;

  Part(this.id, this.serialNumber, this.manufacturer, this.partName, this.image,
      this.price, this.description);

  factory Part.fromJson(Map<String, dynamic> json) => _$PartFromJson(json);

  Map<String, dynamic> toJson() => _$PartToJson(this);
}
